core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -119,23 +147,61 @@ oppia.factory('NumberWithUnitsObjectFactory', [
             value = rawInput.substr(0, ind).trim();
             units = rawInput.substr(ind).trim();
           }
+
+          var keys = Object.keys(CURRENCY_UNITS);
+          for (var i = 0; i < keys.length; i++) {
+            for (var j = 0;
+              j < CURRENCY_UNITS[keys[i]].front_units.length; j++) {
+              if (units.indexOf(
+                CURRENCY_UNITS[keys[i]].front_units[j]) !== -1) {
+                throw new Error(
+                  NUMBER_WITH_UNITS_PARSING_ERRORS.INVALID_CURRENCY_FORMAT);
+              }
+            }
+          }
         } else {
-          if (!rawInput.startsWith('$ ') && !rawInput.startsWith('Rs ') &&
-            !rawInput.startsWith('₹ ')) {
+          var startsWithCurrencyUnits = false;
```

Comment:
Maybe you won't need this loop? You can simply get the index of [0-9] and make sure that unit from beginning to the index of first digit is correct currency unit (which you are verifying in second loop below). Then this one becomes unnecessary.